rasterio/errors.py

```diff
@@ -1,9 +1,15 @@
 """A module of errors."""
 
+from click import FileError
+
 
 class RasterioIOError(IOError):
     """A failure to open a dataset using the presently registered drivers."""
 
 
 class RasterioDriverRegistrationError(ValueError):
-    """To be raised when, eg, _gdal.GDALGetDriverByName("MEM") returns NULL"""
+    """To be raised when, eg, _gdal.GDALGetDriverByName("MEM") returns NULL."""
+
+
+class FileOverwriteError(FileError):
```

Comment:
@sgillies The error message passed to this exception is not properly displayed because `click.FileError()` behaves a little differently than a normal exception.  Right now the error looks like:

```python
Traceback (most recent call last):
  File "buh.py", line 5, in <module>
    print(resolve_inout(files=['LICENSE.txt']))
  File "/Users/kwurster/code/rasterio/rasterio/rio/helpers.py", line 91, in resolve_inout
    "Output exists and won't be overwritten without the "
rasterio.errors.FileOverwriteError: unknown error
```

But modifying the call to the parent `__init__` provides the intended message:

```python
class FileOverwriteError(FileError):
    """Rasterio's CLI refuses to implicitly clobber output files."""

    def __init__(self, message):
        super(FileOverwriteError, self).__init__('', hint=message)
```

```python
Traceback (most recent call last):
  File "buh.py", line 5, in <module>
    print(resolve_inout(files=['LICENSE.txt']))
  File "/Users/kwurster/code/rasterio/rasterio/rio/helpers.py", line 91, in resolve_inout
    "Output exists and won't be overwritten without the "
rasterio.errors.FileOverwriteError: Output exists and won't be overwritten without the `-f` or `-o` options.
```